### PR TITLE
sdpa: Enforce requires_grad=False for attn_mask

### PR DESCRIPTION
**Pull request by galexite**
Requiring grad for the attention mask worked for the `MATH` backend because autograd was able to compute the grad naturally from the decomposition. However, this won't work for other SDPA backends, because there is no explicit grad output for the attention mask in their backward implementations.

Ensure the attention mask does not have `requires_grad=True` at the composite op level for all backends, update the sample inputs to not generate an attention mask requiring grad and add a sample input with a boolean `attn_mask` too.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01

### CODE REVIEW

**Comment by drisspg** [comment]
Where do we systematically assume that all floating point tensors require grad? This make sit so that you can no longer pass in an attn_bias that requires grad
